### PR TITLE
Nits on NSGA2

### PR DESCRIPTION
**Pull request by jonpsy**
Hi @say4n , I've been reading your code on NSGA2 and I really like it :).  I've made some small corrections. 

- minor grammar fix
- replace std::endl with "\n"

### CODE REVIEW

**Comment by ShikharJ** [comment]
Replacing `std::endl` with `\n` is a useless change.

**Comment by jonpsy** [comment]
In hindsight, yes, but I'd rather avoid them if possible. Anycase, these changes are minor nitpicks.

File: include/ensmallen_bits/nsga2/nsga2_impl.hpp [hunk 1]
```diff
@@ -278,7 +278,7 @@ template<typename MatType>
 inline void NSGA2::Crossover(MatType& childA,
                              MatType& childB,
                              const MatType& parentA,
-                             const MatType& parentB)
+                             const MatType&  parentB)
```
Comment (rcurtin) [inline include/ensmallen_bits/nsga2/nsga2_impl.hpp:281]: Seems like it was correct before?

**Review by rcurtin — COMMENTED** [review]
Hey @jonpsy, thanks for the comment fix! :+1: mlpack and ensmallen code tend to use `std::endl` throughout, so we probably ought to stick with that.

Did you have more changes you were planning to make, or is this ready to go?  (If you plan on changing more, we can wait until you're done to finish the review process. :))

File: include/ensmallen_bits/nsga2/nsga2_impl.hpp (hunk 1 above)
Comment (zoq) [inline include/ensmallen_bits/nsga2/nsga2_impl.hpp:281]:
```suggestion
                             const MatType& parentB)
```

Let me provide a quick patch for that :+1:

**Review by zoq — COMMENTED** [review]
Nice catch, even if that one is a small fix, I think it's important to have correct and clear comments :+1:

**Comment by jonpsy** [comment]
Thanks for the reviews. The primary goal of this PR was to get acquainted to the reviewing process. I was making the first pass at this code so I caught a few nitpicks which I've posted.I do realize my changes were not substantial. 

I've made some more changes, but I'd like your opinions before I make them (more than just grammar fix :) ).

I'd like this code to be

- Minimize use of ```fo```  loops
- Write cache-friendly code
- Reduce code length
- Possibly structure it to inherit from abstract class "MultiObjective" so that it gives space to add MOEA/D (my next work)

**Comment by jonpsy** [comment]
There's also a prospect of parallelizing through [OMP](https://ieeexplore.ieee.org/document/6968779).

**Comment by zoq** [comment]
> Thanks for the reviews. The primary goal of this PR was to get acquainted to the reviewing process. I was making the first pass at this code so I caught a few nitpicks which I've posted.I do realize my changes were not substantial.
> 
> I've made some more changes, but I'd like your opinions before I make them (more than just grammar fix :) ). Can we do it on a **seperate** PR? Can we merge this one for the time being?
> 
> I'd like this code to be
> 
> * Minimize use of `for`  loops
> * Write cache-friendly code
> * Reduce code length
> * Possibly structure it to inherit from abstract class "MultiObjective" so that it gives space to add MOEA/D (my next work)

Let's continue the discussion in #263.

**Review by zoq — APPROVED** [review]
Looks good to me.

File: include/ensmallen_bits/nsga2/nsga2_impl.hpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ typename MatType::elem_type NSGA2::Optimize(
   if (lowerBound.n_rows == 1)
     lowerBound = lowerBound(0, 0) * arma::ones(iterate.n_rows, iterate.n_cols);
 
-  // Check if lower bound is a vector of a single dimension.
+  // Check if upper bound is a vector of a single dimension.
```
Comment (say4n) [inline include/ensmallen_bits/nsga2/nsga2_impl.hpp:78]: Oops, thanks, great catch! :)

Comment (jonpsy) [reply include/ensmallen_bits/nsga2/nsga2_impl.hpp:78]: @say4n, Hi, thanks. I'd love your reviews on the follow-up PR. Cheers! :)

**Comment by zoq** [comment]
Thanks for the contribution.

**Pull request merged**